### PR TITLE
Added trailing 0 to ipv6 ranges that end in ":"

### LINKED ISSUE #263: [BUG] Using the range from-to syntax, some IPv6 addresses fail
**Describe the bug**
The `"range"` configuration parameter allows addresses in a from-to format, e.g.

```
"range": "2001:d3ad:b33f:f00d:d00d:d00d:000:0000-2001:d3ad:b33f:f00d:d00d:d00d:0000:0004/64",
```

Which would allocate 5 IPv6 addresses. Some IPv6 addresses however, wind up parsing into a name key for the CRD names that are invalid. With the above range resulting in an error such as:

```
Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_samplepod_default_3f6249c4-191e-44ea-9ef3-f5d8b5d707a6_0(a2dd58c84fcf5069c52b2c229d66639f1e8fd510f1b8a05058e25d89666944fd): error adding pod default_samplepod to CNI network "multus-cni-network": plugin type="multus" name="multus-cni-network" failed (add): [default/samplepod/3f6249c4-191e-44ea-9ef3-f5d8b5d707a6:wb-rangeerror-conf]: error adding container to network "wb-rangeerror-conf": error at storage engine: OverlappingRangeIPReservation.whereabouts.cni.cncf.io "2001-d3ad-b33f-f00d-d00d-d00d--" is invalid: metadata.name: Invalid value: "2001-d3ad-b33f-f00d-d00d-d00d--": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

**Expected behavior**
Parses properly.

**To Reproduce**
Steps to reproduce the behavior, use this set of yaml:

```
---
apiVersion: "k8s.cni.cncf.io/v1"
kind: NetworkAttachmentDefinition
metadata:
  name: macvlan-conf
spec:
  config: '{
	"cniVersion": "0.3.1",
	"name": "wb-rangeerror-conf",
	"plugins": [{
		"type": "macvlan",
		"master": "eth0",
		"mode": "bridge",
		"ipam": {
			"type": "whereabouts",
			"range": "2001:d3ad:b33f:f00d:d00d:d00d:000:0000-2001:d3ad:b33f:f00d:d00d:d00d:0000:0004/64",
			"gateway": "2001:d3ad:b33f:f00d::1"
		}
	}]
}'
---
apiVersion: v1
kind: Pod
metadata:
  name: samplepod
  annotations:
    k8s.v1.cni.cncf.io/networks: macvlan-conf
spec:
  containers:
  - name: samplepod
    command: ["/bin/ash", "-c", "trap : TERM INT; sleep infinity & wait"]
    image: alpine
```

**Environment**:
- Whereabouts version : master
- Kubernetes version (use `kubectl version`): v1.23.6
- OS (e.g. from /etc/os-release): Fedora 35

### LINKED ISSUE #293: Wherabouts IPV6: Range fails for valid IPv6 format [ fdef:b041:5422:0005::]
**Description**

IPv6 addresses ending with all 0's can be represented in the following format with double semicolon ending
fdef:b041:5422:0005::

However when the same is used in the range config in the whereabouts IPAM NAD, the ipam add function fails.

**Range:**
 "range": "fdef:b041:5422:0005::-fdef:b041:5422:0005:7fff:ffff:ffff:ffff/64" 

**Failure:**
 Warning  FailedCreatePodSandBox  116s (x16 over 2m26s)  kubelet            (combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "9c677251030e4fa299cc56b43699c299e437720a1ce9cc73ed61a303460ceba2": plugin type="multus" name="multus-cni-network" failed (add): [efevice-fv-namespace/efevice-fv-20221209-4vxgjbeqnwqpq7gx-d8f6489b7-mp46c:bgf-media-ipv6-net]: error adding container to network "bgf-media-ipv6-net": failed to set up IPAM plugin type "whereabouts": Error at storage engine: OverlappingRangeIPReservation.whereabouts.cni.cncf.io **"fdef-b041-5422-5--" is invalid:** metadata.name: Invalid value: "fdef-b041-5422-5--": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*') 

Whereas, when providing the starting IP as fdef:b041:5422:0005::**0**, there are no issues.
"range": "fdef:b041:5422:0005::0-fdef:b041:5422:0005:7fff:ffff:ffff:ffff/64" 

**Expected behavior**

fdef:b041:5422:0005:: is a valid IPv6 IP. The expectation is for whereabouts to accept that same as range start.

**To Reproduce**
1. Create a whereabouts IPAM NAD with range with start IP in the following format fdef:b041:5422:0005:: [Ending with double colon]
2.  Create a pod using the NAD
3. Check the kubelet logs for failure

**Environment**:
- Whereabouts version : v0.5.4
- Kubernetes version (use `kubectl version`): v1.25.3
- Network-attachment-definition: 
```yaml
apiVersion: "k8s.cni.cncf.io/v1"
kind: NetworkAttachmentDefinition
metadata:
  name: bgf-media-ipv6-net
  namespace: default
  annotations:
    k8s.v1.cni.cncf.io/resourceName: ovs-cni.network.kubevirt.io/br_data
    k8s.v1.cni.cncf.io/nodeSelector: vnf=bgf
spec:
  config: |
    {
      "cniVersion": "0.4.0",
      "plugins": [
        {
          "type": "ovs",
          "bridge": "br_data",
          "vlan": 1804,
          "mtu": 9000,
          "ipam": {
            "type": "whereabouts",
            "range": "fdef:b041:5422:0005::-fdef:b041:5422:0005:7fff:ffff:ffff:ffff/64"
          }
        }
      ]
    }
```
- Whereabouts configuration (on the host): N/A
- OS (e.g. from /etc/os-release): 
~> sudo cat /etc/os-release
NAME="SLES"
VERSION="15-SP4"
VERSION_ID="15.4"
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP4"
ID="sles"
ID_LIKE="suse"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:15:sp4"
DOCUMENTATION_URL="https://documentation.suse.com/"

- Kernel (e.g. `uname -a`): 
~> uname -a
Linux cp-bm25d3 5.14.21-150400.24.38-default #1 SMP PREEMPT_DYNAMIC Fri Dec 9 09:29:22 UTC 2022 (e9c5676) x86_64 x86_64 x86_64 GNU/Linux

- Others: N/A

**Additional info / context**
Add any other information / context about the problem here.

### PR DESCRIPTION
**Pull request by nicklesimba**
**What this PR does / why we need it**:
ipv6 ranges ending in a colon, such as a trail of zeroes denoted by "::" result in the storage engine failing to process the range, because ranges are expected to begin and end with alphanumerical characters to meet CRD requirements. Thus we append a 0 to any ranges that meet this criteria because it is functionally equivalent and meets the requirements.

Signed-off-by: nicklesimba <simha.nikhil@gmail.com>
**Which issue(s) this PR fixes**:
Fixes #263, fixes #293

**Special notes for your reviewer** *(optional)*:
https://imgs.xkcd.com/comics/networking_problems.png

### CODE REVIEW

**Review by dougbtv — CHANGES_REQUESTED** [review]
Totally on board with this change! Can we add a test that ends with a `::`?

Add another test near this one / like this one: https://github.com/k8snetworkplumbingwg/whereabouts/blob/master/cmd/whereabouts_test.go#L223-L235

File: cmd/whereabouts_test.go [hunk 1]
```diff
@@ -234,6 +234,20 @@ var _ = Describe("Whereabouts operations", func() {
 		)
 	})
 
+	It("allocates and releases an IPv6 range that ends with zeroes with a Kubernetes backend", func() {
+
+		ipVersion := "6"
+		ipRange := "2001:1b74:480:603d:0304:0403:000:0000-2001:1b74:480:603d:0304:0403:0000:0004/64"
```
Comment (dougbtv) [inline cmd/whereabouts_test.go:240]: Let's use a documentation range, so make the range start with ` 2001:db8`

Also, add an additional test that doesn't just ends with zeros, but another one that ends with `::` literally so, for example.... `2001:db8:b33f::/48` or something

**Review by dougbtv — CHANGES_REQUESTED** [review]

File: cmd/whereabouts_test.go (hunk 1 above)
Comment (dougbtv) [reply cmd/whereabouts_test.go:240]: see also: https://www.rfc-editor.org/rfc/rfc3849.txt

**Comment by dougbtv** [comment]
Looks good, I think we just need a rebase (there's a merge commit from master in here)

**Review by dougbtv — APPROVED** [review]
LGTM!

**Comment by maiqueb** [comment]
wait, this merged a golang upgrade to 1.19. 

Do we want to do that ? 

Not saying I'm against it, but we probably should have merged https://github.com/k8snetworkplumbingwg/whereabouts/pull/294 instead first.

**Comment by adilGhaffarDev** [comment]
@dougbtv can we cherry-pick this fix to v0.6?

**Pull request merged**